### DISCUSSION TITLE
I really don't know how to solve this issue after ask GPT many times

### DISCUSSION THREAD

**Discussion by gettingalot** [openai/openai-python#1197] [Q&A]
context: i have tried different ways of getting latest version etc. 
:( still not getting it right

import PyPDF2
import openai
from pptx import Presentation

__title__ = "openai"
__version__ = "1.13.2"  # x-release-please-version

# Step 1: Extract text from PDF for all pages
def extract_text_from_pdf(pdf_path):
    text = ""
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        for page in reader.pages:
            text += page.extract_text()
    return text

# Step 2: Summarize text using OpenAI API
def summarize_text(text):
    # Set up OpenAI API
    openai.api_key = 'API_File'
    # Make the API call to summarize
    response = openai.Completion.create(
        model="text-davinci-003",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": text},
        ]
    )
    return response['choices'][0]['message']['content']

# Step 3: Create PowerPoint slides
def create_ppt(summaries):
    prs = Presentation()
    slide_layout = prs.slide_layouts[1]  # Choose slide layout (e.g., Title and Content)
    for summary in summaries:
        slide = prs.slides.add_slide(slide_layout)
        title = slide.shapes.title
        content = slide.placeholders[1]
        title.text = "Summary"
        content.text = summary
    prs.save("summary_slides.pptx")

# Main function
def main(pdf_path):
    # Step 1: Extract text from PDF for all pages
    pdf_text = extract_text_from_pdf(pdf_path)

    # Step 2: Split text into pages and summarize each page
    pages = pdf_text.split('\n\n')  # Split by page separator
    summaries = []
    for page in pages:
        summary = summarize_text(page)
        summaries.append(summary)

        # Step 3: Create PowerPoint slides
    create_ppt(summaries)

if __name__ == "__main__":
    pdf_path = r"C:\Users\Chao\Desktop\2023-gore-street-energy-storage-fund-esg-and-sunstainability-report.pdf"
    main(pdf_path)

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
and i got 
C:\Users\xxx\PycharmProjects\Learning\pythonProject\.venv\Scripts\python.exe C:\Users\xxx\PycharmProjects\Learning\pythonProject\App.py 
Traceback (most recent call last):
  File "C:\Users\xxx\PycharmProjects\Learning\pythonProject\App.py", line 64, in <module>
    main(pdf_path)
  File "C:\Users\xxx\PycharmProjects\Learning\pythonProject\App.py", line 55, in main
    summary = summarize_text(page)
              ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\xxx\PycharmProjects\Learning\pythonProject\App.py", line 23, in summarize_text
    response = openai.Completion.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\xxx\PycharmProjects\Learning\pythonProject\.venv\Lib\site-packages\openai\lib\_old_api.py", line 39, in __call__
    raise APIRemovedInV1(symbol=self._symbol)
openai.lib._old_api.APIRemovedInV1: 

You tried to access openai.Completion, but this is no longer supported in openai>=1.0.0 - see the README at https://github.com/openai/openai-python for the API.

You can run `openai migrate` to automatically upgrade your codebase to use the 1.0.0 interface. 

Alternatively, you can pin your installation to the old version, e.g. `pip install openai==0.28`

A detailed migration guide is available here: https://github.com/openai/openai-python/discussions/742


Process finished with exit code 1

**Comment by rm2631** [comment]
You seem to be using version 0.28's code on a more recent version of OpenAI. Either adapt your code to the most recent SDK version or downgrade to 0.28
